### PR TITLE
Release v4.3.1

### PR DESCRIPTION
**Pull request by tynandebold**
## Description

Release for v4.3.1.

## Development notes

I have updated the release notes as well as the package version.

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Review by studioswong — APPROVED** [review]

**Comment by studioswong** [comment]
@tynandebold please also add a note for the changes in [this PR](https://github.com/kedro-org/kedro-viz/pull/718) on the release notes

**Review by rashidakanchwala — APPROVED** [review]

File: demo-project/lightsail.json [hunk 1]
```diff
@@ -12,4 +12,4 @@
     "containerName": "kedro-viz-live-demo",
     "containerPort": 4141
   }
-}
+}
```
Comment (rashidakanchwala) [inline demo-project/lightsail.json:15]: minor - just need to add the newline

**Comment by tynandebold** [comment]
> @tynandebold please also add a note for the changes in [this PR](https://github.com/kedro-org/kedro-viz/pull/718) on the release notes ( I know this isn't user facing but would help with context regarding this release)

I hear you, though I'm not sure it would help the context in this instance. Ivan had me remove these because of the way our AWS users and keys were generated and shared. We changed that and then took out those lines in that PR. This seems a bit too related to our infrastructure setup and doesn't have much to do with the Viz library itself.

The release we're doing here is really only focused on the websocket connection being able to communicate over `https`, which Lim wrote in the release notes.

**Review by yetudada — APPROVED** [review]

**Comment by studioswong** [comment]
> I hear you, though I'm not sure it would help the context in this instance. Ivan had me remove these because of the way our AWS users and keys were generated and shared. We changed that and then took out those lines in that PR. This seems a bit too related to our infrastructure setup and doesn't have much to do with the Viz library itself.
> 
> The release we're doing here is really only focused on the websocket connection being able to communicate over `https`, which Lim wrote in the release notes.

We do mention changes to our CI setup in our release notes - for this we don't need to specifically mention the generation of AWS users and keys, just a simple `update demo deployment setup on CI` would do.  ( in fact, for release 4.3.0, we should also mention this task on the release notes - something like `setup new demo deployment on CI` would do.)

**Pull request merged**